README.md

```diff
@@ -4,49 +4,250 @@
 
 TypeScript decorators for Polymer 2.0.
 
-### Installation
-
-- Install the decorators:
-  ```sh
-  bower install --save Polymer/polymer-decorators
-  ```
-
-- Import the decorator library in your component definitions:
-  ```html
-  <link rel="import" href="/bower_components/polymer-decorators/polymer-decorators.html">
-  ```
-
-- Include the decorator type declarations in one of the source files in your
-  TypeScript project (be sure to update the reference below with the correct
-  relative path):
-  ```ts
-  /// <reference path="../bower_components/polymer-decorators/global.d.ts" />
-  ```
-
-- Enable the
-  [`experimentalDecorators`](https://www.typescriptlang.org/docs/handbook/decorators.html#metadata)
-  TypeScript compiler setting. Use the `--experimentalDecorators` flag, or
-  update your `tsconfig.json` to include:
-  ```js
-  {
-    "compilerOptions": {
-      "experimentalDecorators": true
-    }
+## Contents
+
+- [Installation](#installation)
+- [Decorator reference](#decorator-reference)
+   - [@customElement](#customelement)
+   - [@property](#property)
+   - [@computed](#computed)
+   - [@observe](#observe)
+   - [@query](#query)
+   - [@queryAll](#queryall)
+   - [@listen](#listen)
+- [Metadata Reflection API](#metadata-reflection-api)
+- [Polymer 1.0](#polymer-10)
+
+
+## Installation
+
+1. Install the decorators with Bower (NPM support coming with Polymer 3.0):
+
+   ```sh
+   bower install --save Polymer/polymer-decorators
+   ```
+
+2. Import the decorator library in your component definitions:
+
+   ```html
+   <link rel="import" href="/bower_components/polymer-decorators/polymer-decorators.html">
+   ```
+
+3. Include type declarations for Polymer (available as of version 2.4) and the
+   Polymer decorators in one your TypeScript source files. You can also add
+   them as sources in your `tsconfig.json` with `include` or `files`.
+
+   ```ts
+   /// <reference path="./bower_components/polymer/types/polymer-element.d.ts" />
+   /// <reference path="./bower_components/polymer-decorators/global.d.ts" />
+   ```
+
+4. Enable the
+   [`experimentalDecorators`](https://www.typescriptlang.org/docs/handbook/decorators.html#metadata)
+   TypeScript compiler setting. Use the `--experimentalDecorators` flag, or
+   update your `tsconfig.json` to include:
+
+   ```js
+   {
+     "compilerOptions": {
+       "experimentalDecorators": true
+     }
+   }
+   ```
+
+5. Optionally [configure Metadata Reflection](#metadata-reflection-api) to define
+   properties more concisely.
+
+
+## Decorator reference
+
+The decorator functions are defined on the `Polymer.decorators` global
+namespace object. You can refer to them directly (e.g.
+`@Polymer.decorators.customElement()`), or you may prefer to assign them to
+shorter variables:
+
+```ts
+const {customElement, property} = Polymer.decorators;
+```
+
+### `@customElement`
+
+Define a custom element.
+
+`tagname` is the name to register this element with. If omitted, the static
```

Comment:
Seems like we should show the signature before detailing the parameters. Maybe the header is a good place. This would also highlight that these are decorator factories, not decorators:

### `@customElement(tagname: string)`

`tagname` is the name to register this element with. If omitted, the static...